servers/src/main/java/tachyon/worker/block/BlockLockManager.java

```diff
@@ -0,0 +1,226 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.worker.block;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.locks.Lock;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Sets;
+
+import tachyon.Constants;
+
+/**
+ * Handle all block locks.
+ * <p>
+ * This class is thread-safe.
+ */
+public class BlockLockManager {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+  /** The number of locks, larger value leads to finer locking granularity, but more space. */
+  private static final int NUM_LOCKS = 100;
+  /** The unique id of each lock */
+  private static final AtomicLong LOCK_ID_GEN = new AtomicLong(0);
+
+  private final BlockMetadataManager mMetaManager;
+  /** A map from a block ID to its lock */
+  private final List<ClientRWLock> mLockArray = new ArrayList<ClientRWLock>(NUM_LOCKS);
```

Comment:
Why is this a list and not an ArrayList (or even an array)? This array should have random access, and also it is called "Array".